cmd/apg/migrate-database.go

```diff
@@ -0,0 +1,143 @@
+// Code generated. DO NOT EDIT.
+
+package main
+
+import (
+	"github.com/spf13/cobra"
+
+	"fmt"
+
+	"github.com/golang/protobuf/jsonpb"
+
+	"os"
+
+	rpcpb "github.com/apigee/registry/rpc"
+)
+
+var MigrateDatabaseInput rpcpb.MigrateDatabaseRequest
+
+var MigrateDatabaseFromFile string
+
+var MigrateDatabaseFollow bool
+
+var MigrateDatabasePollOperation string
+
+func init() {
+	AdminServiceCmd.AddCommand(MigrateDatabaseCmd)
+
+	MigrateDatabaseCmd.Flags().StringVar(&MigrateDatabaseInput.Kind, "kind", "", "A string describing the kind of migration to...")
+
+	MigrateDatabaseCmd.Flags().StringVar(&MigrateDatabaseFromFile, "from_file", "", "Absolute path to JSON file containing request payload")
+
+	MigrateDatabaseCmd.Flags().BoolVar(&MigrateDatabaseFollow, "follow", false, "Block until the long running operation completes")
+
+	AdminServiceCmd.AddCommand(MigrateDatabasePollCmd)
+
+	MigrateDatabasePollCmd.Flags().BoolVar(&MigrateDatabaseFollow, "follow", false, "Block until the long running operation completes")
+
+	MigrateDatabasePollCmd.Flags().StringVar(&MigrateDatabasePollOperation, "operation", "", "Required. Operation name to poll for")
+
+	MigrateDatabasePollCmd.MarkFlagRequired("operation")
+
+}
+
+var MigrateDatabaseCmd = &cobra.Command{
+	Use:   "migrate-database",
+	Short: "MigrateDatabase attempts to migrate the database...",
+	Long:  "MigrateDatabase attempts to migrate the database to the current schema.",
+	PreRun: func(cmd *cobra.Command, args []string) {
+
+		if MigrateDatabaseFromFile == "" {
+
+		}
+
+	},
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+
+		in := os.Stdin
+		if MigrateDatabaseFromFile != "" {
+			in, err = os.Open(MigrateDatabaseFromFile)
+			if err != nil {
+				return err
+			}
+			defer in.Close()
+
+			err = jsonpb.Unmarshal(in, &MigrateDatabaseInput)
+			if err != nil {
+				return err
+			}
+
+		}
+
+		if Verbose {
+			printVerboseInput("Admin", "MigrateDatabase", &MigrateDatabaseInput)
+		}
+		resp, err := AdminClient.MigrateDatabase(ctx, &MigrateDatabaseInput)
+		if err != nil {
+			return err
+		}
+
+		if !MigrateDatabaseFollow {
+			var s interface{}
+			s = resp.Name()
+
+			if OutputJSON {
+				d := make(map[string]string)
+				d["operation"] = resp.Name()
+				s = d
+			}
+
+			printMessage(s)
+			return err
+		}
+
+		result, err := resp.Wait(ctx)
```

Comment:
What's the default timeout on the WaitOperation? Are we considering a flag on customizing waiting time?

Reply:
I don't know -- Here's the source for the CLI generator https://github.com/googleapis/gapic-generator-go/tree/master/cmd/protoc-gen-go_cli.

Reply:
I haven't tried to use the migrate functionality yet, so not sure how long would it takes on the blocking behavior. This should be a good start, and we can add a new flag if we find use cases that need customizing waiting time in the future